### PR TITLE
[Model][Experimental ]Support DeepSeek-V3 Pre-train with MTP

### PR DESCRIPTION
**Pull request by lxd-cumt**
# Description
🎉 We are excited to release the DeepSeek-V3 Pretraining! This PR includes a 15B Model Configuration,  training scripts and module.

The training pipeline has been validated, and key features are as follows
Support Checkpoint Convert (Huggingface -> Megatron) for finetune training
Support MLA, shared experts, sigmoid router, loss-free load balancing based on newest megatron
Support Parallel Strategies, including DataParallel, TensorParallel, PipelineParallel and ExpertParallel
Support MTP

# Work in progress
FP8 training
DualPipe Schedule
Overlap & Memory Optimization
Support for More Chips

### CODE REVIEW

File: examples/deepseek_v3/conf/config_deepseek_v3.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+defaults:
+  - _self_
+  - train: _train_deepseek_v3.yaml
```
Comment (Caozhou1995) [inline examples/deepseek_v3/conf/config_deepseek_v3.yaml]: Remove the underscore _ when naming the file.

File: megatron/megatron/training/arguments.py [hunk 2]
```diff
@@ -451,10 +452,14 @@ def validate_args(args, defaults={}):
...
     map_dtype = lambda d: d if isinstance(d, torch.dtype) else dtype_map[d]
 
-    args.main_grads_dtype = map_dtype(args.main_grads_dtype)
-    args.main_params_dtype = map_dtype(args.main_params_dtype)
-    args.exp_avg_dtype = map_dtype(args.exp_avg_dtype)
-    args.exp_avg_sq_dtype = map_dtype(args.exp_avg_sq_dtype)
+    if args.main_grads_dtype in dtype_map.keys():
+        args.main_grads_dtype = dtype_map[args.main_grads_dtype]
+    if args.main_params_dtype in dtype_map.keys():
+        args.main_params_dtype = dtype_map[args.main_params_dtype]
```
Comment (heavyrain-lzy) [inline megatron/megatron/training/arguments.py:458]: Why change here?

File: megatron/megatron/core/transformer/moe/router.py [hunk 3]
```diff
@@ -112,8 +116,9 @@ def __init__(self, config: TransformerConfig) -> None:
                 torch.zeros(self.config.num_moe_experts, dtype=torch.float32),
                 persistent=False,
             )
+            expert_bias = self.score_bias.data
             self.register_buffer(
-                'expert_bias', torch.zeros(self.config.num_moe_experts, dtype=torch.float32)
+                'expert_bias', expert_bias
```
Comment (heavyrain-lzy) [inline megatron/megatron/core/transformer/moe/router.py:121]: Why change the initial bias value?

**Review by aoyulong — COMMENTED** [review]
It would be beneficial to enhance the pull request description to make it clearer and more informative.

**Review by zhaoyinglia — COMMENTED** [review]
Organize two prs, one for the base module and another for the MTP module.
